src/utils/__tests__/narrow-test.js

```diff
@@ -219,101 +219,59 @@ describe('SEARCH_NARROW', () => {
 describe('isMessageInNarrow', () => {
   const ownEmail = eg.selfUser.email;
 
-  test('any message is in "Home"', () => {
-    const message = eg.streamMessage({ flags: [] });
-    const narrow = HOME_NARROW;
-    expect(isMessageInNarrow(message, narrow, ownEmail)).toBe(true);
-  });
-
-  test('message with type "private" is in private narrow if recipient matches', () => {
-    const message = eg.pmMessage({ flags: [] });
-    const narrow = privateNarrow(eg.otherUser.email);
-    expect(isMessageInNarrow(message, narrow, ownEmail)).toBe(true);
-  });
-
-  test('message to self is in "private" narrow with self', () => {
-    const message = eg.pmMessage({
-      flags: [],
-      display_recipient: [eg.displayRecipientFromUser(eg.selfUser)],
+  // prettier-ignore
+  for (const [narrowDescription, narrow, cases] of [
+    ['all-messages ("home") narrow', HOME_NARROW, [
+      ['a message', true, eg.streamMessage()],
+    ]],
+
+    ['whole-stream narrow', streamNarrow(eg.stream.name), [
+      ['matching stream message', true, eg.streamMessage()],
+    ]],
+    ['stream conversation', topicNarrow(eg.stream.name, 'cabbages'), [
```

Comment:
I've been using "conversation" to refer to the smallest unit of Zulip threading: either a group PM thread, or a topic within a stream. Every conversation is a narrow, but it's a particularly handy subset of narrows in part because every message belongs to exactly one conversation. It's also a term that works pretty naturally in user-facing text.

The same terminology appears in a few places around the project, though I don't think we've standardized on it. One is in the three-dots menu on a message in the webapp, where there's an option "Copy link to conversation". That's for any message; and it'll produce a link narrowed to the message's conversation in this sense, even if you're seeing it from some interleaved narrow.

Reply:
Ah OK, got it! Makes sense.